### PR TITLE
(DOCSP-19905): Fix realm JS SDK versions update Node Tutorial user custom data access, and eslint for web tutorial 

### PR DESCRIPTION
**Pull request by mongodben**
## Pull Request Info

This PR was initiated by this issue - https://github.com/mongodb/docs-realm/issues/1498 

While addressing the issue by bumping the Realm JS version in the web tutorial and across the docs-realm repo, a couple of other issues were encountered and addressed: 
- the Node tutorial `getProjects()` function for getting custom user data was updated to use the `refreshCustomData()` method and have error handling for slow to fire triggers. 
- changed the ESLint version of the web tutorial to be compatible with Create React App

### Jira

- https://jira.mongodb.org/browse/DOCSP-19905 

### Staged Changes (Requires MongoDB Corp SSO)

- [NodeJS CLI Tutorial](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/fix_realm_js_versions/tutorial/nodejs-cli/)

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/tutorial/generated/node/projects.codeblock.getProjects.js [hunk 1]
```diff
@@ -1,7 +1,15 @@
 async function getProjects() {
-  const realm = await index.getRealm(`user=${users.getAuthedUser().id}`);
-  const currentUser = users.getAuthedUser().id;
-  const user = realm.objectForPrimaryKey("User", currentUser);
-  const projects = user.memberOf;
-  return projects;
+  const user = users.getAuthedUser();
+  try {
```
Comment (cbush) [inline source/tutorial/generated/node/projects.codeblock.getProjects.js:3]: Not sure if this really needs a try/catch block. What about returning an empty array in the 'warning' case?

**Review by cbush — APPROVED** [review]

File: source/tutorial/generated/node/projects.codeblock.getProjects.js [hunk 2]
```diff
@@ -1,7 +1,15 @@
...
+    const { memberOf: projects } = await user.refreshCustomData();
+
+    // Make sure that the user object has been created
+    if (!projects) {
+      output.error("The user object hasn't been created yet. Try again soon.");
+      throw new Error("No projects for user");
+    }
+    return projects;
+  } catch (err) {
+    output.error("There was a problem accessing custom user data");
```
[the remark this answers is not in the document]
Comment (mongodben) [reply source/tutorial/generated/node/projects.codeblock.getProjects.js:13]: sure, only helps to add. adding another `output.error(err)` before the one that's already there

File: source/tutorial/generated/node/projects.codeblock.getProjects.js (hunk 1 above)
Comment (mongodben) [reply source/tutorial/generated/node/projects.codeblock.getProjects.js:3]: returning an empty array would just result in an error downstream in the function `selectProject`, so i'd rather catch the error here. not the most robust error handling, but aligns with what's already being done in the project, where errors are presented and the program must be exited. think ok for now, though certainly something that could be improved upon in future iterations.

Comment (mongodben) [reply source/tutorial/generated/node/projects.codeblock.getProjects.js:3]: also including the try/catch catches error returned by the `refreshCustomData()` function in the event that there is no custom user data yet and it throws an error, which tom on the JS team believes to be the outcome if it's called before custom user data is created.

Comment (cbush) [reply source/tutorial/generated/node/projects.codeblock.getProjects.js:3]: By catching you're just returning undefined. And that has to be handled by the caller. Is that being done?

Comment (mongodben) [reply source/tutorial/generated/node/projects.codeblock.getProjects.js:3]: i see what you're saying. as understand the code in this project, this issue of unhandled errors persists persists across the whole project. 

to remedy, you think it'd make sense to have a error handler function that can be used in catch statements across the project?

Comment (cbush) [reply source/tutorial/generated/node/projects.codeblock.getProjects.js:3]: Why not just let the exception go unhandled and bubble up to the top of the program, thereby killing it?

Comment (mongodben) [reply source/tutorial/generated/node/projects.codeblock.getProjects.js:3]: that would certainly work too, and be the easiest way of going about this. created this ticket to address the issue - https://jira.mongodb.org/browse/DOCSP-19916

**Pull request merged**